### PR TITLE
WIP: Feature: Revisions to commands.

### PR DESCRIPTION
**Pull request by iNewLegend**
Depends on https://github.com/elementor/elementor/pull/11993 
Depends on https://github.com/elementor/elementor/pull/12326
Depends on https://github.com/elementor/elementor/pull/12820

### CODE REVIEW

File: modules/history/assets/js/revisions/commands/preview.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+				status: 'autosave',
+				onSuccess: () => {
+					tab.getRevisionViewData( view );
+				},
+			} );
+		} else {
+			tab.getRevisionViewData( view );
+		}
+
+		tab.currentPreviewItem = view;
```
Comment (iNewLegend) [inline modules/history/assets/js/revisions/commands/preview.js:38]: move currents from 'tab' to 'compnent'

File: modules/history/assets/js/revisions/commands/preview.js [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+		if ( revisionID === tab.currentPreviewId ) {
+			return;
+		}
+
+		if ( tab.currentPreviewItem ) {
+			tab.currentPreviewItem.$el.removeClass( 'elementor-revision-current-preview elementor-revision-item-loading' );
+		}
+
+		view.$el.addClass( 'elementor-revision-current-preview elementor-revision-item-loading' );
```
Comment (iNewLegend) [inline modules/history/assets/js/revisions/commands/preview.js:22]: UI HOOK?

Comment (iNewLegend) [inline modules/history/assets/js/revisions/commands/preview.js:22]: HOOK UI?

Comment (iNewLegend) [reply modules/history/assets/js/revisions/commands/preview.js]: https://github.com/elementor/elementor/pull/11994/commits/d67a298dfda6105842695a61048df26214cbdf38

File: modules/history/assets/js/revisions/panel/tab.js [hunk 3]
```diff
@@ -26,13 +26,11 @@ module.exports = Marionette.CompositeView.extend( {
 	document: null,
 
 	initialize: function( options ) {
-		this.document = options.document;
+		options.tab = this;
 
-		this.collection = this.document.revisions.getItems();
+		$e.run( 'panel/history/revisions/initialize', options );
 
 		this.listenTo( elementor.channels.editor, 'saved', this.onEditorSaved );
```
Comment (iNewLegend) [inline modules/history/assets/js/revisions/panel/tab.js:33]: Move to hook.
```suggestion
		this.listenTo( elementor.channels.editor, 'saved', this.onEditorSaved );
```

File: modules/history/assets/js/revisions/commands/index.js [hunk 4]
```diff
@@ -1,2 +1,5 @@
+export { Apply } from './apply';
 export { Down } from './down';
+export { Preview } from './preview';
 export { Up } from './up';
+
```
Comment (iNewLegend) [inline modules/history/assets/js/revisions/commands/index.js:5]: Test

File: modules/history/assets/js/revisions/commands/internal/index.js [hunk 5]
```diff
@@ -0,0 +1 @@
+export { Initialize } from './initialize';
```
Comment (iNewLegend) [inline modules/history/assets/js/revisions/commands/internal/index.js:1]: Test

File: modules/history/assets/js/revisions/hooks/index.js [hunk 6]
```diff
@@ -1 +1,4 @@
-export { RevisionsAfterSave } from './data/save';
+export { RevisionsUpdate } from './data/document/save/save/revisions-update';
+export { RevisionsUpdateCurrentAfter } from './ui/panel/histroy/revisions/preview/revisions-update-current-after';
+export { RevisionsUpdateCurrentBefore } from './ui/panel/histroy/revisions/preview/revisions-update-current-before';
```
Comment (iNewLegend) [inline modules/history/assets/js/revisions/hooks/index.js:3]: test

File: modules/history/assets/js/revisions/component.js [hunk 7]
```diff
@@ -1,8 +1,28 @@
...
+	/**
+	 * @type {Document}
+	 */
+	currentDocument;
+
+	/**
+	 * Temporary.
+	 * @type {RevisionsTabView}
+	 */
+	tab;
```
Comment (iNewLegend) [inline modules/history/assets/js/revisions/component.js:16]: Remove.

**Comment by iNewLegend** [comment]
Use DATA API instead of ajax.

**Comment by iNewLegend** [comment]
DO NOT REMOVE
modules/history/assets/js/revisions/panel/tab.js

+ Fix current selected revision after closing revisions tab.

**Comment by matipojo** [comment]
It will be merged in the future.

**Pull request closed without merging**